cpp/src/arrow/pretty_print-test.cc

```diff
@@ -64,36 +73,82 @@ void Check(const T& obj, const PrettyPrintOptions& options, const char* expected
 }
 
 template <typename TYPE, typename C_TYPE>
-void CheckPrimitive(int indent, const std::vector<bool>& is_valid,
+void CheckPrimitive(const PrettyPrintOptions& options, const std::vector<bool>& is_valid,
                     const std::vector<C_TYPE>& values, const char* expected) {
   std::shared_ptr<Array> array;
   ArrayFromVector<TYPE, C_TYPE>(is_valid, values, &array);
-  CheckArray(*array, indent, expected);
+  CheckArray(*array, options, expected);
 }
 
 TEST_F(TestPrettyPrint, PrimitiveType) {
   std::vector<bool> is_valid = {true, true, false, true, false};
 
   std::vector<int32_t> values = {0, 1, 2, 3, 4};
-  static const char* expected = R"expected([0, 1, null, 3, null])expected";
-  CheckPrimitive<Int32Type, int32_t>(0, is_valid, values, expected);
-
-  std::vector<std::string> values2 = {"foo", "bar", "", "baz", ""};
-  static const char* ex2 = R"expected(["foo", "bar", null, "baz", null])expected";
-  CheckPrimitive<StringType, std::string>(0, is_valid, values2, ex2);
+  static const char* expected = "[\n  0,\n  1,\n  NA,\n  3,\n  NA\n]";
```

Comment:
I think we ought to use `null` or `NULL` for nulls by default -- it even might be better to use this in Python. I originally used `NA` as a nod to R but I think it introduces some domain-specific ideas that don't really fit (we use "null" to describe this concept everywhere else in the codebase)

We can also make this configurable in `PrettyPrintOptions`

Reply:
We could add an `enum` or a `std::string` in `PrettyPrintOptions`. @wesm Feel free to do what you think works best.